File: compiler/bench/treecount.c
// expected: 262143
#include <stdio.h>
long treecount(long depth) {
    if (depth == 0) return 1;
    return treecount(depth - 1) + treecount(depth - 1) + 1;
}

int main() {
    for (int i = 0; i < 1000; i++) {
        treecount(17);
    }
    printf("%ld\n", treecount(17));
    return 0;
}

File: compiler/bench/treecount.fs
\ expected: 262143
\ Count nodes recursively, simulate 100K node tree (depth 17)

: treecount ( depth -- count ) recursive
  dup 0= if drop 1 exit then
  1- dup recurse swap recurse + 1+ ;

: main
  1000 0 do 17 treecount drop loop
  17 treecount . cr ;
